src/apply.c

```diff
@@ -4312,4 +4316,37 @@ flip_through_book(struct obj *obj)
     return ECMD_TIME;
 }
 
+static int
+flip_coin(struct obj *obj)
+{
+    struct obj *otmp = obj;
+    boolean lose_coin = FALSE;
+
+    You("flip %s.", an(xname(obj)));
+    if (Underwater) {
+        pline_The("%s floats away.", xname(obj));
+        lose_coin = TRUE;
+    } else if (Glib || Fumbling || (ACURR(A_DEX) < 10 && !rn2(ACURR(A_DEX)))) {
+        pline_The("%s slips between your %s.", xname(obj),
+              fingers_or_gloves(FALSE));
+        lose_coin = TRUE;
+    }
+
+    if (lose_coin) {
+        if (otmp->quan > 1L)
+            otmp = splitobj(otmp, 1L);
+        dropx(otmp);
+        return ECMD_TIME;
+    }
+    if (Hallucination) {
+        pline(rn2(100) ? "Wow, a double header!"
+                        /* edge case */
+                       : "The coin miraculously lands on its edge!");
+    } else {
+        pline_The("%s comes up %s.", xname(obj), 
```

Comment:
Most of the time the messages produced will be "You flip a gold piece. The gold piece comes up heads/tails." Changing it to "It comes up heads/tails" would flow better.

(Also, if you apply multiple gold pieces, wouldn't this produce "The 247 gold pieces comes up heads/tails"?)

Reply:
Applying multiple gold pieces does not result in odd messaging. Tested. 

Agreed on the use of a pronoun.

Reply:
As far as throwing a gold piece upwards goes, I think xnethack's solution is very elegant, as it usually is. In my mind though, flipping a coin is a distinct action involving manipulating a coin with one's fingers, not throwing a coin upward. That may be a meaningless distinction, however.